src/components/ThemeToggle/ThemeToggle.tsx

```diff
@@ -21,18 +21,12 @@ const darkTheme = (
 )
 
 const ThemeToggle = () => {
-  const { data } = useGetThemeQuery()
-  const { theme, setTheme } = useTheme()
+  const { setTheme } = useTheme()
   const { trackEvent } = useAnalytics()
+  const { portalUser } = useAuthContext()
   const { user } = useUser()
   const [handleEditThemeMutation] = useEditThemeMutation()
 
-  useEffect(() => {
-    if (data) {
-      setTheme(data.theme)
-    }
-  }, [data])
-
   // This is necessary to avoid a rehydration error since we render server side
   // See this blog for details: https://www.joshwcomeau.com/react/the-perils-of-rehydration/
   if (!user) {
```

Comment:
The portalUser type does have userId on it, but the getUser query does not retrieve it, so it isn't on the object here. I tried to update the query to also return userId but I kept getting an error saying that it wasn't allowed. Since we already have the userId in the session, I tried to just pass it into the portalUser manually whenever it is initialized but that only worked sometimes.

With that being said, I decided to leave this in for now. I think we can execute useUser in each layout component just like we're now doing with getUser and place that info into the context there. Then we can remove the numerous calls to the hook that exist in various components and utilize the context. That seemed like a little bit too much to tack onto this story though, so figured I'd leave it.